### ISSUE TITLE
Is it possible to speed things up, perhaps by running the tests in parallel?

### ISSUE THREAD

**Issue by cb-sarathkumar** [sitespeedio/sitespeed.io#4286]
### Your question

My regression suite currently takes about 2 hours to run, and we're exploring options to reduce this execution time. Is it possible to speed things up, perhaps by running the tests in parallel?

**Comment by soulgalore** [comment]
Hi @cb-sarathkumar you can run the tests and split them on multiple machines? If you run on the same machine, you need to make sure that they do not interfere with each other (like multiple Chrome/Firefox instances reaches memory/CPU limits etc).

**Comment by cb-sarathkumar** [comment]
Here’s a polished response you could use:

---

Thanks for the response! I can definitely run the tests on multiple machines. If I use the `--multi` tag, will it execute the tests in parallel?

We're using the following file as a runner for the entire suite:

`export default async function regressionRunner(context, commands) {
  await subscriptionRunner(context, commands);
  await customerRunner(context, commands);
  await checkoutRunner(context, commands);
}`

If I run this file with the `--multi` flag, will the tests within it run in parallel?

**Comment by soulgalore** [comment]
> If I run this file with the --multi flag, will the tests within it run in parallel?.

No. Multi in this context means multiple pages in a user journey.

I would recommend running on multiple servers and then send the result to Graphite (or if you have another time series database) to aggregate the result. Running tests on the same server and keeping track if they disturb each other is complicated and needs a lot of work. Thinking you want to avoid spending time/money on investigating regression that happened because you run multiple tests on the same machine.
